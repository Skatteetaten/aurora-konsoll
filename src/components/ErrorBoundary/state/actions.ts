import { IAppError, IErrorState } from 'models/StateManager/ErrorStateManager';
import { createAction } from 'redux-ts-utils';

const errorBoundaryAction = (action: string) => `errorBoundary/${action}`;

export const fetchErrorsResponse = createAction<Response>(
  errorBoundaryAction('FETCHED_ERRORS_RESPONSE')
);

export const currentErrors = createAction<IErrorState>(
  errorBoundaryAction('CURRENT_ERRORS')
);

export const currentErrorAction = createAction<IAppError | undefined>(
  errorBoundaryAction('CURRENT_ERROR')
);

export default {
  fetchErrorsResponse,
  currentErrors,
  currentErrorAction
};
