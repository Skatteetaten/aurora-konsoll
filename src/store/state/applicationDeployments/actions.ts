import { ActionType } from 'typesafe-actions';

import {
  IApplicationsConnectionData,
  IApplicationDeploymentWithDetailsData,
} from 'services/auroraApiClients/applicationDeploymentClient/query';
import { IDataAndErrors } from 'services/GoboClient';
import { createAsyncActions } from 'utils/redux/action-utils';
import { createAction } from 'redux-ts-utils';
import { ApplicationDeployment } from 'models/immer/ApplicationDeployment';

const action = (action: string) => `applicationDeployments/${action}`;

const fetchApplicationDeployments = createAsyncActions<
  IDataAndErrors<IApplicationsConnectionData>
>(action('REQUEST_APPLICATIONS'));

const fetchApplicationDeploymentWithDetails = createAsyncActions<
  IDataAndErrors<IApplicationDeploymentWithDetailsData>
>(action('REQUEST_APPLICATION_DEPLOYMENT_WITH_DETAILS'));

const refreshApplicationDeployment = createAction<void>(
  action('REQUEST_REFRESH_APPLICATION_DEPLOYMENT')
);

const refreshAllDeploymentsForAffiliation = createAction<void>(
  action('REQUEST_REFRESH_DEPLOYMENT_FOR_AFFILIATION')
);

const deployRequest = createAction<void>(action('DEPLOY_REQUEST'));

const resetApplicationDeploymentState = createAction<void>(
  action('RESET_APPLICATION_DEPLOYMENT_STATE')
);

const deleteApplicationDeploymentRequest = createAction<void>(
  action('DELETE_APPLICATION_DEPLOYMENT_REQUEST')
);

const setApplicationDeployment = createAction<ApplicationDeployment>(
  action('SET_APPLICATION_DEPLOYMENT')
);

export const actions = {
  deployRequest,
  fetchApplicationDeployments,
  refreshAllDeploymentsForAffiliation,
  fetchApplicationDeploymentWithDetails,
  resetApplicationDeploymentState,
  refreshApplicationDeployment,
  deleteApplicationDeploymentRequest,
  setApplicationDeployment,
};

export type ApplicationsAction = ActionType<typeof actions>;
