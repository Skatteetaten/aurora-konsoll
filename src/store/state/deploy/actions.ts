import { createAction } from 'redux-ts-utils';
import { ActionType } from 'typesafe-actions';

const action = (action: string) => `deploy/${action}`;

const isDeploying = createAction<{ inProgress: boolean; version: string }>(
  action('DEPLOY')
);
const deployResult = createAction<{ version: string; success: boolean }>(
  action('DEPLOY_SUCCESS')
);

export const actions = {
  isDeploying,
  deployResult
};

export type DeployAction = ActionType<typeof actions>;
