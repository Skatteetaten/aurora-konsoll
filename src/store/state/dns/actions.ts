import { ActionType } from 'typesafe-actions';

import { IDataAndErrors } from 'services/GoboClient';
import { createAsyncActions } from 'utils/redux/action-utils';
import { CnameInfosQuery } from 'services/auroraApiClients/dnsClient/query';

const action = (action: string) => `dns/${action}`;

const fetchCnameInfosRequest = createAsyncActions<
  IDataAndErrors<CnameInfosQuery>
>(action('FETCH_CNAME_INFOS'));

export const actions = {
  fetchCnameInfosRequest,
};

export type DnsAction = ActionType<typeof actions>;
