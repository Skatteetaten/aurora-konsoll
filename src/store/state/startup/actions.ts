import { createAsyncActions } from 'utils/redux/action-utils';
import { IDataAndErrors } from 'services/GoboClient';
import { IUserAndAffiliationsData } from 'services/auroraApiClients/applicationDeploymentClient/query';

const action = (action: string) => `currentUser/${action}`;

const [
  requestCurrentUser,
  requestCurrentUserSuccess,
  requestCurrentUserFailure
] = createAsyncActions<IDataAndErrors<IUserAndAffiliationsData>>(
  action('REQUEST_CURRENT_USER')
);

export const actions = {
  requestCurrentUser,
  requestCurrentUserSuccess,
  requestCurrentUserFailure
};
