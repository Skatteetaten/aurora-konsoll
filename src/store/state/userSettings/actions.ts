import { ActionType } from 'typesafe-actions';
import { createAsyncActions } from 'utils/redux/action-utils';
import { IGoboResult } from 'services/GoboClient';
import { IUserSettingsData } from 'services/auroraApiClients/userSettingsClient/query';

const action = (action: string) => `userSettings/${action}`;

const [
  requestUserSettings,
  requestUserSettingsSuccess,
  requestUserSettingsFailure
] = createAsyncActions<void, IGoboResult<IUserSettingsData>>(
  action('REQUEST_USER_SETTINGS')
);

const [
  requestUpdateUserSettings,
  requestUpdateUserSettingsSuccess,
  requestUpdateUserSettingsFailure
] = createAsyncActions<void, IGoboResult<{ updateUserSettings: boolean }>>(
  action('REQUEST_UPDATE_USER_SETTINGS')
);

export const actions = {
  requestUserSettings,
  requestUserSettingsSuccess,
  requestUserSettingsFailure,
  requestUpdateUserSettings,
  requestUpdateUserSettingsSuccess,
  requestUpdateUserSettingsFailure
};

export type UserSettingsAction = ActionType<typeof actions>;
