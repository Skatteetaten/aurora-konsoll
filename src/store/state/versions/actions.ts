import { createAction } from 'redux-ts-utils';
import { ImageTagType } from 'models/ImageTagType';
import { ITagsQuery } from 'services/auroraApiClients/imageRepositoryClient/query';
import { ActionType } from 'typesafe-actions';
import { createAsyncActions } from 'utils/redux/action-utils';
import { IDataAndErrors } from 'services/GoboClient';

const action = (action: string) => `versions/${action}`;

const fetchInitVersions = createAsyncActions<
  Record<ImageTagType, IDataAndErrors<ITagsQuery>>
>(action('FETCH_INIT_VERSIONS'));

const fetchVersionsForType = createAsyncActions<{
  response: IDataAndErrors<ITagsQuery>;
  type: ImageTagType;
  paged: boolean;
}>(action('FETCH_VERSIONS_FOR_TYPE'));

const resetState = createAction<void>(action('RESET'));

const resetStateForType = createAction<ImageTagType>(action('CLEAR_TYPE'));

export const actions = {
  fetchInitVersions,
  fetchVersionsForType,
  resetState,
  resetStateForType
};

export type VersionsAction = ActionType<typeof actions>;
