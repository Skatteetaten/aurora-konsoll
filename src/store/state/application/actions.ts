import { StateThunk } from 'store/types';

import { actions } from './actions.type';

export const deploy = (
  applicationDeploymentId: string,
  version: string
): StateThunk => async (dispatch, getState, { clients }) => {
  const client = clients.applicationDeploymentClient;

  dispatch(actions.isDeploying({ inProgress: true, version }));
  const response = await client.redeployWithVersion(
    applicationDeploymentId,
    version
  );
  dispatch(actions.isDeploying({ inProgress: false, version }));

  if (response) {
    dispatch(
      actions.deployResult({
        version,
        success: response.data.redeployWithVersion
      })
    );
    if (response.errors) {
      // Dispatch error
    }
  } else {
    // Dispatch error
  }
};
