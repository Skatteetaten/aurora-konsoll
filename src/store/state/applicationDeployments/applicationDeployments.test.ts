import { GraphQLSeverMock } from 'utils/GraphQLMock';
import { createTestStore } from 'utils/redux/test-utils';
import { applicationDeploymentFactory } from 'testData/testDataBuilders';

import { actions } from './actions';
import {
  fetchApplicationDeployments,
  refreshAllDeploymentsForAffiliation,
  deleteAndRefreshApplications,
} from './action.creators';
import {
  IApplicationDeploymentData,
  IApplicationsConnectionData,
} from 'services/auroraApiClients/applicationDeploymentClient/query';

const server = new GraphQLSeverMock();

const {
  nextAction,
  dispatch,
  clearActionsAndResetState,
  skipActions,
  getActionQueue,
} = createTestStore(server);

function createApplicationConnectionData(
  deployments: IApplicationDeploymentData[]
): { data: IApplicationsConnectionData } {
  return {
    data: {
      applications: {
        edges: [
          {
            node: {
              name: 'mokey',
              applicationDeployments: deployments,
            },
          },
        ],
      },
    },
  };
}

beforeAll(() => {
  server.putResponse(
    'getApplicationDeployments',
    createApplicationConnectionData(
      applicationDeploymentFactory.buildList(1, {
        time: '2019-11-06T12:54:00.707621Z',
      })
    )
  );

  server.putResponse('refreshApplicationDeployments', {
    data: {
      refreshApplicationDeployments: true,
    },
  });

  server.putResponse('deleteApplicationDeployment', {
    data: {
      deleteApplicationDeployment: true,
    },
  });
});

afterAll(() => {
  server.close();
});

afterEach(() => {
  clearActionsAndResetState();
});

describe('applicationDeployments', () => {
  it('should fetch applications', async () => {
    await dispatch(fetchApplicationDeployments(['aurora']));

    nextAction(
      actions.fetchApplicationDeployments.request,
      ({ applications }) => {
        expect(applications).toMatchSnapshot();
      }
    );

    nextAction(
      actions.fetchApplicationDeployments.success,
      ({ applications }) => {
        expect(applications).toMatchSnapshot();
      }
    );

    expect(getActionQueue()).toHaveLength(0);
  });

  it('should refresh and then fetch applications', async () => {
    // Adding application deployments to state
    await dispatch(fetchApplicationDeployments(['aurora']));
    skipActions(getActionQueue().length);

    // Simulate response after refresh with updated cache time.
    server.putResponse(
      'getApplicationDeployments',
      createApplicationConnectionData(
        applicationDeploymentFactory.buildList(1, {
          time: '2019-11-06T12:56:00.707621Z',
        })
      )
    );

    await dispatch(refreshAllDeploymentsForAffiliation('aurora'));

    nextAction(
      actions.refreshAllDeploymentsForAffiliation,
      ({ applications }) => {
        expect(applications).toMatchSnapshot();
        expect(applications.applicationsConnection.getCacheTime()).toBe(
          '2019-11-06T12:54:00.707621Z'
        );
      }
    );

    nextAction(
      actions.fetchApplicationDeployments.success,
      ({ applications }) => {
        expect(applications).toMatchSnapshot();
        expect(applications.applicationsConnection.getCacheTime()).toBe(
          '2019-11-06T12:56:00.707621Z'
        );
      }
    );

    expect(getActionQueue()).toHaveLength(0);
  });

  it('should delete application deployment and then fetch applications', async () => {
    // Adding application deployments to state
    await dispatch(fetchApplicationDeployments(['aurora']));
    skipActions(getActionQueue().length);

    // Simulate response after delete with no applications
    server.putResponse(
      'getApplicationDeployments',
      createApplicationConnectionData([])
    );

    await dispatch(
      deleteAndRefreshApplications('aurora', 'aurora-dev', 'mokey')
    );

    nextAction(
      actions.deleteApplicationDeploymentRequest,
      ({ applications }) => {
        expect(applications).toMatchSnapshot();
      }
    );

    nextAction(
      actions.fetchApplicationDeployments.success,
      ({ applications }) => {
        expect(applications).toMatchSnapshot();
      }
    );

    expect(getActionQueue()).toHaveLength(0);
  });
});
