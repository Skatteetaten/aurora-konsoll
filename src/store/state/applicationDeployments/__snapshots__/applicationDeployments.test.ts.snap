// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applicationDeployments should delete application deployment and then fetch applications 1`] = `
Object {
  "applicationsConnection": ApplicationsConnection {
    "applicationDeployments": Array [
      Object {
        "affiliation": "aurora",
        "environment": "dev",
        "id": "1",
        "imageRepository": Object {
          "guiUrl": "http://localhost",
          "repository": "no.skatteetaten.aurora.mokey",
        },
        "message": undefined,
        "name": "mokey",
        "namespace": "aurora-dev",
        "permission": Object {
          "paas": Object {
            "admin": true,
            "view": true,
          },
        },
        "status": Object {
          "code": "HEALTHY",
          "reasons": Array [],
          "reports": Array [],
        },
        "time": "2019-11-06T12:56:00.707621Z",
        "version": Object {
          "auroraVersion": undefined,
          "deployTag": Object {
            "lastModified": "",
            "name": "latest",
            "type": "LATEST",
          },
          "releaseTo": undefined,
        },
      },
    ],
    Symbol(immer-draftable): true,
  },
  "errors": Object {
    "requestApplicationDeployment": Array [],
    "requestApplications": Array [],
  },
  "isDeploying": false,
  "isFetching": true,
  "isRefreshing": false,
  "isRefreshingForAffiliation": true,
}
`;

exports[`applicationDeployments should delete application deployment and then fetch applications 2`] = `
Object {
  "applicationsConnection": ApplicationsConnection {
    "applicationDeployments": Array [],
    Symbol(immer-draftable): true,
  },
  "errors": Object {
    "requestApplicationDeployment": Array [],
    "requestApplications": Array [],
  },
  "isDeploying": false,
  "isFetching": false,
  "isRefreshing": false,
  "isRefreshingForAffiliation": false,
}
`;

exports[`applicationDeployments should fetch applications 1`] = `
Object {
  "applicationsConnection": ApplicationsConnection {
    "applicationDeployments": Array [],
    Symbol(immer-draftable): true,
  },
  "errors": Object {
    "requestApplicationDeployment": Array [],
    "requestApplications": Array [],
  },
  "isDeploying": false,
  "isFetching": true,
  "isRefreshing": false,
  "isRefreshingForAffiliation": false,
}
`;

exports[`applicationDeployments should fetch applications 2`] = `
Object {
  "applicationsConnection": ApplicationsConnection {
    "applicationDeployments": Array [
      Object {
        "affiliation": "aurora",
        "environment": "dev",
        "id": "0",
        "imageRepository": Object {
          "guiUrl": "http://localhost",
          "repository": "no.skatteetaten.aurora.mokey",
        },
        "message": undefined,
        "name": "mokey",
        "namespace": "aurora-dev",
        "permission": Object {
          "paas": Object {
            "admin": true,
            "view": true,
          },
        },
        "status": Object {
          "code": "HEALTHY",
          "reasons": Array [],
          "reports": Array [],
        },
        "time": "2019-11-06T12:54:00.707621Z",
        "version": Object {
          "auroraVersion": undefined,
          "deployTag": Object {
            "lastModified": "",
            "name": "latest",
            "type": "LATEST",
          },
          "releaseTo": undefined,
        },
      },
    ],
    Symbol(immer-draftable): true,
  },
  "errors": Object {
    "requestApplicationDeployment": Array [],
    "requestApplications": Array [],
  },
  "isDeploying": false,
  "isFetching": false,
  "isRefreshing": false,
  "isRefreshingForAffiliation": false,
}
`;

exports[`applicationDeployments should refresh and then fetch applications 1`] = `
Object {
  "applicationsConnection": ApplicationsConnection {
    "applicationDeployments": Array [
      Object {
        "affiliation": "aurora",
        "environment": "dev",
        "id": "0",
        "imageRepository": Object {
          "guiUrl": "http://localhost",
          "repository": "no.skatteetaten.aurora.mokey",
        },
        "message": undefined,
        "name": "mokey",
        "namespace": "aurora-dev",
        "permission": Object {
          "paas": Object {
            "admin": true,
            "view": true,
          },
        },
        "status": Object {
          "code": "HEALTHY",
          "reasons": Array [],
          "reports": Array [],
        },
        "time": "2019-11-06T12:54:00.707621Z",
        "version": Object {
          "auroraVersion": undefined,
          "deployTag": Object {
            "lastModified": "",
            "name": "latest",
            "type": "LATEST",
          },
          "releaseTo": undefined,
        },
      },
    ],
    Symbol(immer-draftable): true,
  },
  "errors": Object {
    "requestApplicationDeployment": Array [],
    "requestApplications": Array [],
  },
  "isDeploying": false,
  "isFetching": true,
  "isRefreshing": false,
  "isRefreshingForAffiliation": true,
}
`;

exports[`applicationDeployments should refresh and then fetch applications 2`] = `
Object {
  "applicationsConnection": ApplicationsConnection {
    "applicationDeployments": Array [
      Object {
        "affiliation": "aurora",
        "environment": "dev",
        "id": "1",
        "imageRepository": Object {
          "guiUrl": "http://localhost",
          "repository": "no.skatteetaten.aurora.mokey",
        },
        "message": undefined,
        "name": "mokey",
        "namespace": "aurora-dev",
        "permission": Object {
          "paas": Object {
            "admin": true,
            "view": true,
          },
        },
        "status": Object {
          "code": "HEALTHY",
          "reasons": Array [],
          "reports": Array [],
        },
        "time": "2019-11-06T12:56:00.707621Z",
        "version": Object {
          "auroraVersion": undefined,
          "deployTag": Object {
            "lastModified": "",
            "name": "latest",
            "type": "LATEST",
          },
          "releaseTo": undefined,
        },
      },
    ],
    Symbol(immer-draftable): true,
  },
  "errors": Object {
    "requestApplicationDeployment": Array [],
    "requestApplications": Array [],
  },
  "isDeploying": false,
  "isFetching": false,
  "isRefreshing": false,
  "isRefreshingForAffiliation": false,
}
`;
